vault/models/vault_entry.py

```diff
@@ -71,10 +75,25 @@ def _compute_complete_name(self):
 
     @api.model
     def search_panel_select_range(self, field_name):
-        """Add context to show just the name in searchpanel instead of full path"""
-        return super(
-            VaultEntry, self.with_context(entry_short_name=True)
-        ).search_panel_select_range(field_name)
+        """Add context to show just the name in searchpanel instead of full path.
+        We need to override function to prevent 200 records limit in search panel."""
+        old_value = models.SEARCH_PANEL_LIMIT
+        models.SEARCH_PANEL_LIMIT = False
+        _self = self.with_context(
+            search_panel_domain_override=True, entry_short_name=True
+        )
+        res = super(VaultEntry, _self).search_panel_select_range(field_name)
+        models.SEARCH_PANEL_LIMIT = old_value
+        return res
+
+    def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
+        """We apply a domain to not list records that do not have children."""
+        if self.env.context.get("search_panel_domain_override"):
+            domain = [("child_ids", "!=", False)]
```

Comment:
From a navigational perspective this makes sense to not show the leaves of the tree. I'm not quite sure if you want to fully overwrite it or extend the domain here. It could make a difference if your view/filter is set.